src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/debtposition/InstallmentProcessingService.java

```diff
@@ -0,0 +1,101 @@
+        this.debtPositionService = debtPositionService;
+        this.installmentSynchronizeMapper = installmentSynchronizeMapper;
+        this.installmentErrorsArchiverService = installmentErrorsArchiverService;
+        this.dpInstallmentsWorkflowCompletionService = dpInstallmentsWorkflowCompletionService;
+    }
+
+    /**
+     * Processes a stream of InstallmentIngestionFlowFileDTO and synchronizes each installment.
+     *
+     * @param iterator          Stream of installment ingestion flow file DTOs to be processed.
+     * @param ingestionFlowFile Metadata of the ingestion file containing details about the ingestion process.
+     * @param workingDirectory  The directory where error files will be written if processing fails.
+     * @return An {@link InstallmentIngestionFlowFileResult} containing details about the processed rows, errors, and archived files.
+     */
+    public InstallmentIngestionFlowFileResult processInstallments(Iterator<InstallmentIngestionFlowFileDTO> iterator,
+                                                                  IngestionFlowFile ingestionFlowFile,
+                                                                  Path workingDirectory) {
+        List<InstallmentErrorDTO> errorList = new ArrayList<>();
+        long processedRows = 0;
+        long totalRows = 0;
+
+        while (iterator.hasNext()) {
+            totalRows++;
+
+            InstallmentIngestionFlowFileDTO installment = iterator.next();
+            InstallmentSynchronizeDTO installmentSynchronizeDTO = installmentSynchronizeMapper.map(
+                    installment,
+                    ingestionFlowFile.getIngestionFlowFileId(),
+                    ingestionFlowFile.getOrganizationId()
+            );
+            try {
+                // For the moment we have decided to call the GPD api for a single DP because their development for the massive v2.0 is not yet ready
+                String workflowId = debtPositionService.installmentSynchronize("ORDINARY_SIL", installmentSynchronizeDTO, false);
+                if (workflowId != null) {
+                    boolean workflowCompleted = dpInstallmentsWorkflowCompletionService.waitForWorkflowCompletion(
+                            workflowId, installment, ingestionFlowFile.getFileName(), errorList
+                    );
+                    log.info("Workflow with id {} completed", workflowId);
+                    if (workflowCompleted) {
+                        processedRows++;
+                    }
+                } else {
+                    processedRows++;
+                }
+            } catch (Exception e) {
+                log.error("Error processing installment {}: {}", installment.getIud(), e.getMessage());
+                InstallmentErrorDTO error = dpInstallmentsWorkflowCompletionService.buildInstallmentErrorDTO(
+                        ingestionFlowFile.getFileName(), installment, null, "PROCESS_EXCEPTION", e.getMessage());
```

Comment:
what's the purpose of this method `dpInstallmentsWorkflowCompletionService.buildInstallmentErrorDTO` it's just calling the constructor/builder...
just call the constructor/builder here instead of exposing a method to be reused here...
Otherwise, move the method elsewhere: why should I expect to use a service called DpInstallmentsWorkflowCompletionService to create the ErrorDto?

ps: a methodFactory could have a sense if you have to hide some inner details of creating an object... otherwise let's just call the new